clients/fbc_repository.go

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright (c) 2023 Dell Inc., or its subsidiaries. All Rights Reserved.
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://mozilla.org/MPL/2.0/
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clients
+
+import (
+	"fmt"
+	"terraform-provider-ome/models"
+)
+
+// GetAllRepositories retrieves all repositories from the given catalogs array.
+func (c *Client) GetAllRepositories(catalogsArray []models.CatalogsModel) (*[]models.RepositoryModel, error) {
```

Comment:
This function does not invoke any API calls. Is it possible to move this logic to the helper package?

Reply:
Also, is there any particular reason to use pointers to slices? Go slices are themselves basically pointers to some underlying array and can be passed by value without any significant overhead. So IMHO pointers to slices are best avoided whenever possible.